### ISSUE TITLE
Suggestion: load day data in as data-* attributes

### ISSUE THREAD

**Issue by Abraham-Tourdex** [uxsolutions/bootstrap-datepicker#2432]
I'd like an easy way to roll through the calendar and select days by their date or whatever, so I can style them and inject extra data into the elements themselves. I'm injecting `data-*` attributes manually on `DOMContentLoaded` and on `changeMonth` but I kinda hoped this would already be a feature. Any reason why there is no extended selector data being loaded in to the calendar?

**Comment by omaraboumrad** [comment]
Agreed. Additionally, it would be very helpful if  **beforeShowDay** would pass in the container element (relevant td for that day), so we can manipulate said element the way we see fit (such as adding data attributes).

Instead of monitoring the DOM, I opted to add the data attributes by doing HTML injection in **beforeShowDay** as follows (mega hack):

```javascript
$('#calendar').datepicker({
    beforeShowDay: function(day){
        return {classes: `some-class" data-day="stuff`};
    }
})
```

This will end up with `<td class="day some-class" data-day="stuff">`. Would've been much simpler if `beforeShowDay` received the `element` as a second argument. It would be backwards compatible and wouldn't break the existing tests. Alternatively, allow `beforeShowDay`'s return to also accept an `attrs` that get attached to the element, for example: `{classes: some-class", attrs: {"data-day": "stuff"}};`

My preference is the first option, [which you have hinted at in an old issue.
](https://github.com/uxsolutions/bootstrap-datepicker/issues/1421#issuecomment-113060803)
